Runtime/Timeline/CinemachineMixer.cs

```diff
@@ -25,61 +25,85 @@ internal sealed class CinemachineMixer : PlayableBehaviour
 #if UNITY_EDITOR && UNITY_2019_2_OR_NEWER
         class ScrubbingCacheHelper
         {
-            // Registry of all vcams that are present in the track, active or not
-            List<List<CinemachineVirtualCameraBase>> mAllCamerasForScrubbing;
+            // Remember the active clips of the previous frame so we can track camera cuts
+            public int ActivePlayableA;
+            public int ActivePlayableB;
+
+            struct ClipObjects
+            {
+                public List<List<CinemachineVirtualCameraBase>> Cameras;
+                public float MaxDampTime;
+            }
+            List<ClipObjects> CachedObjects;
+
+            static List<CinemachineVirtualCameraBase> scratch = new List<CinemachineVirtualCameraBase>();
 
             public void Init(Playable playable)
             {
                 // Build our vcam registry for scrubbing updates
-                mAllCamerasForScrubbing = new List<List<CinemachineVirtualCameraBase>>();
+                CachedObjects = new List<ClipObjects>(playable.GetInputCount());
                 for (int i = 0; i < playable.GetInputCount(); ++i)
                 {
+                    var cs = new ClipObjects 
+                    {
+                        Cameras = new List<List<CinemachineVirtualCameraBase>>(),
+                    };
+
                     var clip = (ScriptPlayable<CinemachineShotPlayable>)playable.GetInput(i);
                     CinemachineShotPlayable shot = clip.GetBehaviour();
                     if (shot != null && shot.IsValid)
                     {
-                        var vcam = shot.VirtualCamera;
-                        int nestLevel = 0;
-                        for (ICinemachineCamera p = vcam.ParentCamera; p != null; p = p.ParentCamera)
-                            ++nestLevel;
-                        while (mAllCamerasForScrubbing.Count <= nestLevel)
-                            mAllCamerasForScrubbing.Add(new List<CinemachineVirtualCameraBase>());
-                        if (mAllCamerasForScrubbing[nestLevel].IndexOf(vcam) < 0)
-                            mAllCamerasForScrubbing[nestLevel].Add(vcam);
+                        var mainVcam = shot.VirtualCamera;
+                        cs.Cameras.Add(new List<CinemachineVirtualCameraBase>());
+
+                        // Add all child cameras
+                        scratch.Clear();
+                        mainVcam.GetComponentsInChildren(scratch);
+                        for (int j = 0; j < scratch.Count; ++j)
+                        {
+                            var vcam = scratch[j];
+
+                            int nestLevel = 0;
+                            for (ICinemachineCamera p = vcam.ParentCamera; 
+                                    p != null && p != (ICinemachineCamera)mainVcam; p = p.ParentCamera)
+                                ++nestLevel;
```

Comment:
I would add {} around _++nestLevel;_ here, because of the line breaks, I got disoriented at first sight. :)

Reply:
ok